internal/stacksapi/stack_notification_test.go

```diff
@@ -32,53 +44,78 @@ func TestCreateStackNotification(t *testing.T) {
 		t.Cleanup(func() { server.Close() })
 
-		req := CreateStackNotificationRequest{
+		req := CreateStackNotificationsRequest{
 			StackKey: "stack-123",
-			JobUUID:  "456",
-			Detail:   "Pod is starting up",
+			Notifications: []StackNotification{
+				{
+					JobUUID:   "456",
+					Detail:    "Pod 1 starting",
+					Timestamp: sampleTime,
+				},
+				{
+					JobUUID: "789",
+					Detail:  "Pod 2 starting",
+				},
+			},
 		}
 
-		header, err := client.CreateStackNotification(t.Context(), req)
+		resp, header, err := client.CreateStackNotifications(t.Context(), req)
 		if err != nil {
-			t.Fatalf("client.CreateStackNotification returned an error: %v", err)
+			t.Fatalf("client.CreateStackNotifications returned an error: %v", err)
 		}
 
 		want, got := "custom-value", header.Get("X-Custom-Header")
 		if want != got {
 			t.Errorf("header.Get(\"X-Custom-Header\") = %q, want %q", got, want)
 		}
+
+		assert.Empty(t, resp.Errors)
 	})
 
-	t.Run("crops detail when exceeds 255 characters", func(t *testing.T) {
+	t.Run("returns errors for validation failures", func(t *testing.T) {
 		t.Parallel()
 
 		server, client := setupTestServer(t, func(w http.ResponseWriter, r *http.Request) {
-			verifyAuthMethodPath(t, r, "POST", "/stacks/stack-123/jobs/456/stack_notifications")
-
-			var params CreateStackNotificationRequest
-			assert.NoError(t, json.NewDecoder(r.Body).Decode(&params))
-
-			// Verify the detail was cropped to 255 characters
-			assert.Equal(t, 255, len(params.Detail))
-			assert.True(t, strings.HasSuffix(params.Detail, "… (cropped)"))
+			verifyAuthMethodPath(t, r, "POST", "/stacks/stack-123/notifications")
 
 			w.WriteHeader(http.StatusOK)
+			assert.NoError(t, json.NewEncoder(w).Encode(CreateStackNotificationsResponse{
+				Errors: []StackNotificationError{
+					{
+						Error:   "detail is required",
+						Indexes: []int{1},
+					},
+					{
+						Error:   "detail exceeds its length limit of 256",
+						Indexes: []int{3},
+					},
+				},
+			}))
 		})
 		t.Cleanup(func() { server.Close() })
 
-		// Create a detail longer than 255 characters
-		largeDetail := strings.Repeat("x", 300) // 300 characters
-
-		req := CreateStackNotificationRequest{
+		req := CreateStackNotificationsRequest{
 			StackKey: "stack-123",
-			JobUUID:  "456",
-			Detail:   largeDetail,
+			Notifications: []StackNotification{
+				{JobUUID: "456", Detail: "Valid"},
+				{JobUUID: "789", Detail: ""},
+				{JobUUID: "abc", Detail: "Another valid"},
+				{JobUUID: "def", Detail: string(make([]byte, 257))},
+			},
 		}
 
-		_, err := client.CreateStackNotification(t.Context(), req)
+		resp, _, err := client.CreateStackNotifications(t.Context(), req)
 		assert.NoError(t, err)
+		assert.Len(t, resp.Errors, 2)
+		assert.Equal(t, "detail is required", resp.Errors[0].Error)
+		assert.Equal(t, []int{1}, resp.Errors[0].Indexes)
+		assert.Equal(t, "detail exceeds its length limit of 256", resp.Errors[1].Error)
+		assert.Equal(t, []int{3}, resp.Errors[1].Indexes)
```

Comment:
This would be nicer with `cmp.Diff` (https://google.github.io/styleguide/go/decisions#full-structure-comparisons):

```suggestion
		wantErrors := []StackNotificationError{
			{
				Error:   "detail is required",
				Indexes: []int{1},
			},
			{
				Error:   "detail exceeds its length limit of 256",
				Indexes: []int{3},
			},
		}
		if diff := cmp.Diff(resp.Errors, wantErrors); diff != "" {
			t.Errorf("resp.Errors diff (-got +want):\n%s", diff)
		}
```